server/src/main/java/org/apache/cloudstack/network/topology/BasicNetworkVisitor.java

```diff
@@ -198,7 +198,7 @@ public boolean visit(final DhcpEntryRules dhcp) throws ResourceUnavailableExcept
         final DeployDestination destination = dhcp.getDestination();
         final boolean remove = dhcp.isRemove();
 
-        if (router.getPodIdToDeployIn().longValue() == destination.getPod().getId()) {
+        if (destination != null && router.getPodIdToDeployIn().longValue() == destination.getPod().getId()) {
```

Comment:
Might as well be more defensive and add `router != null && router.getPodIdToDeployIn() != null`. In general being defensive is safer I think.

Reply:
Thanks @anuragaw, added defensive checks

Reply:
@anuragaw @nvazquez 
This would avoid the command being executed at all, which is not the original intention. The destination is not necessary when removing something, as the DhcpRule in this case should already be deployed. So wouldn't it make more sense to change the if to:

```
if (router != null &&
            (remove || (destination != null &&
                                    destination.getPod() != null && router.getPodIdToDeployIn() != null &&
                                    router.getPodIdToDeployIn().longValue() == destination.getPod().getId()))) {
```